File: 101. Symmetric Tree/s.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func isSymmetric(root *TreeNode) bool {
	return isMirror(root.Left, root.Right)
}

func isMirror(p1 *TreeNode, p2 *TreeNode) bool {
	if p1 != nil && p2 != nil {
		return p1.Val == p2.Val && isMirror(p1.Left, p2.Right) && isMirror(p1.Right, p2.Left)
	}
	if p1 == p2 {
		return true
	}
	return false
}

File: 101. Symmetric Tree/s.rs
// Definition for a binary tree node.
#[derive(Debug, PartialEq, Eq)]
pub struct TreeNode {
  pub val: i32,
  pub left: Option<Rc<RefCell<TreeNode>>>,
  pub right: Option<Rc<RefCell<TreeNode>>>,
}

impl TreeNode {
  #[inline]
  pub fn new(val: i32) -> Self {
    TreeNode {
      val,
      left: None,
      right: None
    }
  }
}
use std::rc::Rc;
use std::cell::RefCell;

type PTN = Option<Rc<RefCell<TreeNode>>>;
fn isMirror(left:PTN, right:PTN) -> bool{
  if left == right{
      return true
  }
  return false
}
struct Solution;
impl Solution {
    pub fn is_symmetric(root: Option<Rc<RefCell<TreeNode>>>) -> bool {
      return isMirror(*root.unwrap().left, *root.unwrap().right)
    }
}

fn main(){

}